src/session.h

```diff
@@ -52,6 +52,16 @@ namespace ledger {
 
 class xact_t;
 
+struct ComparePaths
+{
+  bool operator()(const path& p1, const path& p2) const
+  {
+    return p1 < p2 && !boost::filesystem::equivalent(p1, p2);
+  }
+};
+
+#define COMMA ,
```

Comment:
What is the purpose of this macro definition?

Reply:
The definition of `data_files` is done via the macro `OPTION__`. In order to provide an own comparison fuction for `std::set<path>`, we need to provide it as a template parameter which would mean writing `std::set<path, ComparePaths>`. But because this is all part of a macro parameter, the "," gets interpreted by the precompiler and we get an error (something along the lines of "macro `OPTION__` only allows 3 parameters, not 4"). There are several ways to work around this, one is defining a macro which inserts the "," (which I did). Other solutions involve modifying the macro being used (`OPTION__`) and in order to keep things simple, I wanted to avoid that. I just realized that boost already defines such a comma macro in `boost/preprocessor/punctuation/comma.hpp` (`BOOST_PP_COMMA()`) so I guess using that one would be the appropriate way.